### DISCUSSION TITLE
Welcome to mqtt-energenie-ener314rt Discussions!

### DISCUSSION THREAD

**Discussion by Achronite** [Achronite/mqtt-energenie-ener314rt#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us how you are using this application.

**Comment by 43tudor** [comment]
good evening @Achronite i am new to home assistant. i have had energenie products for serveral years now and for many i was happy with basic function from there hub and app. I used to use IFTTT to collect data and allowed some TRV's to follow the Nest controller. Recently i have have purchased a heat pump and solar pannels and found haome assistant. 

to start with i would like to record data from the TRV's then maybe use HA to control them. I followed your various posted but to be honest i am little confused. As the old PC i had HA running on i purchased a pi and the energenie card and was hoping i could just download and away we go. It looks like it a little over my head. are you able to help pleae. 

I am running HA on the pi directly rather than docor.

**Comment by Achronite** [comment]
You need access to a standard shell prompt on the pi to run this, as it needs to run natively in whichever flavour of LINUX/ Raspberry pi OS that you have so that it has access to GPIO / SPI.

> I am running HA on the pi directly rather than docker.

Do you mean you are using Home Assistant Operating System (HAOS) on the pi?  If you are then I don't think you will have this access by default; you will need to use one of the other [install methods](https://www.home-assistant.io/installation/raspberrypi).

**Comment by sgulati-hub** [comment]
Hi @Achronite, I have now captured another commands sent from the MiHome gateway to the thermostat as "REPORT_PERIOD"
The REPORT_PERIOD  correspond to the "Winter Mode Response Time (second)" and "Summer Mode Response time (minute)" in the setting for MiHome Thermostat in the App. (nb: It always sends and stores the STATE as reporting period in seconds i.e. if Thermostat is set on Summer mode in the app then is sends the STATE as "Summer Mode Response time (minute)" * 60 otherwise it sends the response time as "Winter Mode Response Time (second)"
I have verified that it changes the frequency of WAKEUP for the thermostat.

<img src="https://github.com/Achronite/mqtt-energenie-ener314rt/assets/61850770/ca15a246-f6aa-47e2-af09-8ad6758aa2b1" width=400px/>

Is there any quick way I can send this via mqtt-energenie-ener314rt/HomeAssistant and verify?

**Reply by Achronite** [reply]
Excellent @sgulati-hub! 

I've tried and failed to get the REPORT_PERIOD command recorded (did I mention i bought a 2nd hand gateway off ebay?); I think I'll try to have another attempt at capturing it; It's underlying data type and bytes I need to make it work properly; and that is only shown in the verbose debug logs).

What did you do to force it to change the values?  Is it in the app under 'Thermostat settings'?

**Reply by sgulati-hub** [reply]
Yes, it is in the App under "Thermostat settings".  It is consistently capturing for me if I change this value in the app (and save the changes) in the App. 
Is there any way I can capture the logs to get data type and bytes?

**Comment by Achronite** [comment]
Got it!
```
openThings_decode() len=24, decoded data:24:4:18:68:29:224:29:132:203:18:1:32:203:18:1:32:210:4:0:0:0:60:0:160:130: crc:41090, crca:41090, deviceId=14687620 - CRC OK
1713093783:12:18:14687620:0 TARGET_TEMP "id:"203(command),"datatype":2,"str":"","int":288,"float":18.000000}
:1 TARGET_TEMP "id:"203(command),"datatype":2,"str":"","int":288,"float":18.000000}
:2 REPORT_PERIOD "id:"210(command),"datatype":1,"str":"","int":60,"float":60.000000}
```

**Comment by Achronite** [comment]
It's very strange; I've tried sending a REPORT_PERIOD *on it's own* via my code and the thermostat seems to ignore it.... I'm wondering if I always need to accompany it with a TARGET_TEMP in the same message???

I'll have to do some more testing later this week.
